frontend/src/hooks/api/integrations/queries.tsx

```diff
@@ -110,3 +111,28 @@ export const useDeleteIntegration = () => {
     }
   });
 };
+
+export const useSyncIntegration = (pollingRef: MutableRefObject<NodeJS.Timeout | null>) => {
```

Comment:
You can keep the pollingRef timer inside the function and the timer cleanup useEffect also here. 

Then return both the mutation and `isSyncing` variable to show the syncing status thus outside the function nobody needs to care about the timer